Add events on app CRs (#121)
giantswarm/app-admission-controller: docs/events.md

## Emit events

> At the end of successful validation of app CRs, app-admission-controllers check the changes from the following specs and emit each of events.
> - `.sepc.version` 

Reviewer: ```suggestion
At the end of each successful validation of an app CR, app-admission-controller checks the changes for the following fields and emits an event.

- `.spec.version` 
```

- `.spec.version` 

> - `.sepc.catalog`

Reviewer: These ones still have the typo.

- `.sepc.userConfig.secret`
- `.sepc.config.configMap`
- `.sepc.config.secret`

> To see where comparison occurs, see [validate_app.go](../app-admission-controller/pkg/admission/validate_app.go).
> 

Reviewer: ```suggestion
The comparison occurs here [validate_app.go](../app-admission-controller/pkg/admission/validate_app.go).
```

> If you describe the app CR objects in kubectl, you can see the `AppUpdated` events as below.

Reviewer: ```suggestion
If you describe the app CR with kubectl you can see the events e.g.
```

Just making this a bit shorter so it's easier to read.

```
kubectl -n giantswarm describe apps userd-unique

...

Events:
  Type    Reason      Age    From                      Message
  ----    ------      ----   ----                      -------
  Normal  AppUpdated  3m2s   app-admission-controller  version has been changed to `1.1.0`
  Normal  AppUpdated  2m49s  app-admission-controller  version has been changed to `1.1.1`

>   Normal  AppUpdated  1s     app-admission-controller  appConfigMap has been resetted 

Reviewer: ```suggestion
  Normal  AppUpdated  1s     app-admission-controller  appConfigMap has been reset 
```

Sorry I missed this earlier. This can be either reset or resetting to describe the action while it's in progress.
